flexget/plugins/api_tvmaze.py

```diff
@@ -0,0 +1,401 @@
+    summary = Column(Unicode)
+    webchannel = Column(String)
+    runtime = Column(Integer)
+    show_type = Column(String)
+    network = Column(Unicode)
+    seasons = relation('TVMazeSeasons', order_by='TVMazeSeasons.number', cascade='all, delete, delete-orphan',
+                       backref='series')
+    last_update = Column(DateTime)  # last time we updated the db for the show
+
+    def __init__(self, series, session):
+        self.update(series, session)
+
+    def update(self, series, session):
+        self.status = series.status
+        self.rating = series.rating['average']
+        self.weight = series.weight
+        self.updated = datetime.fromtimestamp(series.updated)
+        self.name = series.name
+        self.language = series.language
+        self.schedule = series.schedule
+        self.url = series.url
+        try:
+            self.original_image = series.image.get('original')
+        except AttributeError:
+            self.original_image = None
+        try:
+            self.medium_image = series.image.get('medium')
+        except AttributeError:
+            self.medium_image = None
+        self.tvdb_id = series.externals.get('thetvdb')
+        self.tvrage_id = series.externals.get('tvrage')
+        self.premiered = parser.parse(series.premiered)
+        self.summary = series.summary
+        self.webchannel = series.webChannel
+        self.runtime = series.runtime
+        self.show_type = series.type
+        self.maze_id = series.maze_id
+        self.network = series.network['name']
+        self.last_update = datetime.now()
+
+        self.seasons[:] = get_db_season(self.maze_id, series.seasons, session)
+        self.genres[:] = get_db_genres(series.genres, session)
+
+    def __repr__(self):
+        return '<TVMazeSeries(title=%s,id=%s,last_update=%s)>' % (self.name, self.id, self.last_update)
+
+    def __str__(self):
+        return self.name
+
+    @property
+    def expired(self):
+        if not self.last_update:
+            log.debug('no last update attribute, series set for update')
+            return True
+        time_dif = datetime.now() - self.last_update
+        expiration = time_dif.days > UPDATE_INTERVAL
+        log.debug('series {0} is expired: {1}'.format(self.name, expiration))
+        return expiration
+
+
+class TVMazeSeasons(Base):
+    __tablename__ = 'tvmaze_season'
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    series_maze_id = Column(Integer, ForeignKey('tvmaze_series.maze_id'), nullable=False)
+    number = Column(Integer)
+    episodes = relation('TVMazeEpisodes', order_by='TVMazeEpisodes.season_number', cascade='all, delete, delete-orphan',
+                        backref='season')
+    last_update = Column(DateTime)
+
+    def __init__(self, season, maze_id, session):
+        super(TVMazeSeasons, self).__init__()
+        self.update(season, maze_id, session)
+
+    def update(self, season, maze_id, session):
```

Comment:
Yeah. This is a method on an already cached item, no? In what situation would you want to change the maze_id after you create it?